pilot/pkg/serviceregistry/federation/controller.go

```diff
@@ -512,10 +513,12 @@ func (c *Controller) getEgressServiceAddrs() ([]model.NetworkGateway, []string)
 				Addr: address.IP,
 				Port: uint32(common.DefaultFederationPort),
 			})
-			sas = append(sas, serviceAccountByIP[address.IP])
+			if sa, exists := serviceAccountByIP[address.IP]; exists {
+				uniqueSAs.Insert(sa)
+			}
 		}
 	}
-	return addrs, sas
+	return addrs, uniqueSAs.UnsortedList()
```

Comment:
Given that `UnsortedList()` returns elements in random order (i.e. different order every time you invoke it), can this change now break any processing that happens afterwards? For example, when you have an array stored in a custom resource, and a controller calculates this array during every reconcile, if the order of elements changes, the controller constantly updates the resource, even though semantically there is no real change in the array. 

The previous version of this function always returned the same order, since it was based on the order of the addresses and subsets in the resource, but that's no longer the case. Would returning a sorted list be more appropriate?

Reply:
I didn't take it into account, but it definitely makes sense. Thanks!

Reply:
I fixed it.

Reply:
Fixed